xwiki-rendering-syntaxes/xwiki-rendering-syntax-xhtml/src/main/java/org/xwiki/rendering/internal/parser/xhtml/wikimodel/XWikiCommentHandler.java

```diff
@@ -112,18 +113,12 @@ public void onComment(String content, TagStack stack)
 
     private void handleMacroCommentStart(String content, TagStack stack)
     {
-        boolean shouldIgnoreAll;
 
-        // true if we are already in a non generated content block
-        boolean inNonGeneratedContent = stack.getStackParameter(NON_GENERATED_CONTENT_STACK) != null
-            && (Boolean) stack.getStackParameter(NON_GENERATED_CONTENT_STACK);
+        boolean inNonGeneratedContent = isInNonGeneratedContent(stack);
```

Comment:
Looks like this is only used for `shouldIgnoreAll`, so why not instead create a method to compute `shouldIgnoreAll` value?

Reply:
Why not. But the first question to answer is, how to accurately compute `inNonGeneratedContent`?